ocean_lib/ocean/ocean_assets.py

```diff
@@ -141,24 +141,77 @@ def build_data_tokens_list(
         self, services: list, deployed_erc20_tokens: list
     ) -> list:
         data_tokens = []
+        # (1-n) service per datatoken, 1 datatoken per service
         for erc20_token in deployed_erc20_tokens:
             for service in services:
-                data_token = {
-                    "address": erc20_token.address,
-                    "name": erc20_token.contract.caller.name(),
-                    "symbol": erc20_token.symbol(),
-                    "serviceId": service.id,
-                }
-                data_tokens.append(data_token)
+                if service.data_token == erc20_token.address:
+                    data_tokens.append(
+                        {
+                            "address": erc20_token.address,
+                            "name": erc20_token.contract.caller.name(),
+                            "symbol": erc20_token.symbol(),
+                            "serviceId": service.id,
+                        }
+                    )
         return data_tokens
 
+    @staticmethod
+    def process_ddo(
```

Comment:
I think a better name for this function would be _encrypt_ddo_locally, or _encrypt_locally_if_needed, or even _apply_ddo_flags. It is unclear otherwise what "processing" is happening, without looking at the code